pkg/analyzer/cronjob_test.go

```diff
@@ -15,219 +15,165 @@ package analyzer
 
 import (
 	"context"
+	"sort"
 	"testing"
 
 	"github.com/k8sgpt-ai/k8sgpt/pkg/common"
 	"github.com/k8sgpt-ai/k8sgpt/pkg/kubernetes"
-	"github.com/magiconair/properties/assert"
+	"github.com/stretchr/testify/require"
 	batchv1 "k8s.io/api/batch/v1"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes/fake"
 )
 
-func TestCronJobSuccess(t *testing.T) {
-	clientset := fake.NewSimpleClientset(&batchv1.CronJob{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "example-cronjob",
-			Namespace: "default",
-			Annotations: map[string]string{
-				"analysisDate": "2022-04-01",
-			},
-			Labels: map[string]string{
-				"app": "example-app",
-			},
-		},
-		Spec: batchv1.CronJobSpec{
-			Schedule:          "*/1 * * * *",
-			ConcurrencyPolicy: "Allow",
-			JobTemplate: batchv1.JobTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{
-						"app": "example-app",
-					},
-				},
-				Spec: batchv1.JobSpec{
-					Template: v1.PodTemplateSpec{
-						Spec: v1.PodSpec{
-							Containers: []v1.Container{
-								{
-									Name:  "example-container",
-									Image: "nginx",
-								},
-							},
-							RestartPolicy: v1.RestartPolicyOnFailure,
-						},
-					},
-				},
-			},
-		},
-	})
+func TestCronJobAnalyzer(t *testing.T) {
+	suspend := new(bool)
+	*suspend = true
 
-	config := common.Analyzer{
-		Client: &kubernetes.Client{
-			Client: clientset,
-		},
-		Context:   context.Background(),
-		Namespace: "default",
-	}
+	startingDeadline := new(int64)
+	*startingDeadline = -7
```

Comment:
It would be nice to be able to tell if a variable is for a failure case or not just by its name.

Reply:
Thanks for the suggestion! Cleaner code is always helpful for everyone.